### DISCUSSION TITLE
Fail to open bmp file with Magick.NET.Core 13.1.3

### DISCUSSION THREAD

**Discussion by zaocaoer** [dlemstra/Magick.NET#1402] [Help]
Fail to open bmp file with Magick.NET.Core 13.1.3
http://img.alicdn.com/imgextra/i1/2675528095/O1CN01HJ9TmD29fYy0a7jZU_!!2675528095.bmp
https://img.alicdn.com/imgextra/i1/2209602883215/O1CN011ZMagx1ZcW39v5g5u_!!2209602883215.bmp

ImageMagick.MagickCorruptImageErrorException:“length and filesize do not match `C:/Users/c/AppData/Local/Temp/magick-t1F5r648YvbcwvsuciRA9w1QMWKV3lcb' @ error/bmp.c/ReadBMPImage/852”

**Comment by dlemstra** [comment]
How are you reading the file? And can you please attach the images here? You might need to zip them.

**Reply by zaocaoer** [reply]
```C#
using var image = new MagickImage("https://img.alicdn.com/imgextra/i1/2209602883215/O1CN011ZMagx1ZcW39v5g5u_!!2209602883215.bmp");
```

[bmp.zip](https://github.com/dlemstra/Magick.NET/files/11744694/bmp.zip)

**Reply by dlemstra** [reply]
What happens when you do:
```C#
var settings = new MagickReadSettings(new BmpReadDefines
{
    IgnoreFileSize = true,
});

using var image = new MagickImage("url", settings);
```

**Reply by zaocaoer** [reply]
It is ok to read that bmp by using the BmpReadDefines. 
Is there any side effect to read picture are not bmp by using the BmpReadDefines?

**Reply by dlemstra** [reply]
There are no side effects when you read another format and use the BmpReadDefines.

**Reply by zaocaoer** [reply]
Ok. Thanks.
